org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/admin/NamespacesResource.java

```diff
@@ -150,15 +152,21 @@ public String getPrefixForEncodedNamespace(@PathParam("namespace") String uri) {
 	 */
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
-	public List<NamespaceWithPrefix> getNamespacesAsJSONlist() {
+	public List<NamespaceWithPrefix> getNamespacesAsJSONlist(@QueryParam("all") String allNamespaces) {
 		Collection<Namespace> namespaces = this.getNamespaces();
 
 		// We now have all namespaces
 		// We need to convert from Namespace to String
 
 		List<NamespaceWithPrefix> namespacesList = new ArrayList<>();
 		for (Namespace ns : namespaces) {
-			namespacesList.add(new NamespaceWithPrefix(ns, this.namespaceManager.getPrefix(ns.getDecoded())));
+			if(allNamespaces != null){
+				namespacesList.add(new NamespaceWithPrefix(ns, this.namespaceManager.getPrefix(ns.getDecoded())));
+			}else{
+				if(!Namespaces.getNotAllowedNamespaces().contains(ns.getDecoded())){
+					namespacesList.add(new NamespaceWithPrefix(ns, this.namespaceManager.getPrefix(ns.getDecoded())));	
+				}
```

Comment:
Can you try Java 8 for the whole loop?

if (Objects.isNull(allNamespaces)) {
    namespaceList = namespaces.stream().map(ns -> ns.getDecoded()).filter(ns -> Namespaces.getAllowedNamespaces().contains(ns)).map(ns -> new NamespaceWithPrefix(ns, this.namespaceManager.getPrefix(ns))).collect.Collectors.toList();

} else {
  // allNamespaces is not null, thus all namespaces have to be added without any filterting
  namespaceList = namespaces.stream().map(ns -> new NamespaceWithPrefix(ns, this.namespaceManager.getPrefix(ns.getDecoded()))).collect.Collectors.toList();
}